rusoto/credential/src/profile.rs

```diff
@@ -307,107 +305,51 @@ fn parse_config_file(file_path: &Path) -> Option<HashMap<String, HashMap<String,
 }
 
 /// Parses a Credentials file into a Map of <`ProfileName`, `AwsCredentials`>
-fn parse_credentials_file(
-    file_path: &Path,
-) -> Result<HashMap<String, AwsCredentials>, CredentialsError> {
-    match fs::metadata(file_path) {
-        Err(_) => {
-            return Err(CredentialsError::new(format!(
-                "Couldn't stat credentials file: [ {:?} ]. Non existant, or no permission.",
-                file_path
+fn parse_credentials_file<P>(
+    file_path: P,
+) -> Result<HashMap<String, AwsCredentials>, CredentialsError>
+where
+    P: AsRef<Path>,
+{
+    let profiles: HashMap<_, _> = parse_config_file(&file_path)
+        .map(|data| {
+            Ok(data
+                .into_iter()
+                .filter_map(|(profile, properties)| {
+                    if let (Some(key), Some(secret)) = (
+                        properties.get("aws_access_key_id"),
+                        properties.get("aws_secret_access_key"),
+                    ) {
+                        Some((
+                            profile,
+                            AwsCredentials::new(
+                                key,
+                                secret,
+                                properties
+                                    .get("aws_session_token")
+                                    .or_else(|| properties.get("aws_security_token"))
+                                    .map(String::to_owned),
+                                None,
+                            ),
+                        ))
+                    } else {
+                        None
+                    }
+                })
+                .collect())
+        })
+        .unwrap_or_else(|| {
+            Err(CredentialsError::new(format!(
+                "Invalid credentials file: [ {} ].",
+                file_path.as_ref().display()
```

Comment:
I took the liberty if using the `display()` value of the path object. I think that's what you typically want for user display.